setup/setup.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Main setup script
+SCRIPT_DIR="$(cd "$(dirname "$0")/scripts" && pwd)"
+SCRIPT_LIST_FILE="$(cd "$(dirname "$0")" && pwd)/script-list.txt"
+
+# Display help
+display_help() {
+    echo "Usage: ./setup.sh [COMMAND]"
+    echo ""
+    echo "Commands:"
+    echo "  help       Show this help message and list available scripts"
+    echo "  all        Run all scripts in the order listed in script-list.txt"
+    while IFS= read -r line || [ -n "$line" ]; do
+        # Ignore empty lines and comments
+        [[ "$line" =~ ^#.*$ || -z "$line" ]] && continue
+        script_name=$(echo "$line" | cut -d':' -f1)
+        script_desc=$(echo "$line" | cut -d':' -f2-)
+        echo "  $script_name: $script_desc"
+    done < "$SCRIPT_LIST_FILE"
+}
+
+# Function to run a single script
+run_script() {
+    local script_name="$1"
+    local script_path="$SCRIPT_DIR/$script_name"
+
+    if [ -f "$script_path" ]; then
+        echo "Running script: $script_name"
+        chmod +x "$script_path"
+        "$script_path"
+    else
+        echo "Script not found: $script_name"
+    fi
+}
+
+# Function to run all scripts
+run_all() {
+    echo "Running all scripts in order..."
+    while IFS= read -r line || [ -n "$line" ]; do
+        # Ignore empty lines and comments
+        [[ "$line" =~ ^#.*$ || -z "$line" ]] && continue
+        script_name=$(echo "$line" | cut -d':' -f1)
+        run_script "$script_name"
+    done < "$SCRIPT_LIST_FILE"
+    echo "All scripts executed successfully!"
+}
+
+# Check if script-list.txt exists
+if [ ! -f "$SCRIPT_LIST_FILE" ]; then
```

Comment:
I don't like that this chunk is in the middle of a group of functions. 

I would either make this a function too or move it down to the main loop